src/patternfly/components/Drawer/drawer.scss

```diff
@@ -1,112 +1,241 @@
+  --pf-c-drawer--m-expanded--m-panel-left__panel--BoxShadow: var(--pf-global--BoxShadow--lg-right);
 
   // Divider
-  --pf-c-drawer--m-divider__panel--after--Width: var(--pf-global--BorderWidth--sm);
-  --pf-c-drawer--m-divider--m-expanded__panel--after--BackgroundColor: var(--pf-global--BorderColor--100);
+  --pf-c-drawer__panel--after--Width: var(--pf-global--BorderWidth--sm);
+  --pf-c-drawer__panel--after--BackgroundColor: transparent;
+  --pf-c-drawer--m-inline--m-expanded__panel--after--BackgroundColor: var(--pf-global--BorderColor--100);
 
   display: flex;
+  flex-direction: column;
+  min-width: var(--pf-c-drawer__panel--MinWidth);
   height: 100%;
   overflow-x: hidden;
 }
 
+// Header area
+.pf-c-drawer__section {
+  flex-grow: 0;
+  background-color: var(--pf-c-drawer__section--BackgroundColor);
+
+  &.pf-m-no-background {
+    --pf-c-drawer__section--BackgroundColor: transparent;
+  }
+}
+
+// Main area
+.pf-c-drawer__main {
+  display: flex;
+  flex-grow: 1;
+  overflow: hidden;
+}
+
 // Content / panel
 .pf-c-drawer__content,
 .pf-c-drawer__panel {
+  display: flex;
+  flex-direction: column;
   flex-shrink: 0;
   overflow: auto;
 }
 
 // Content
 .pf-c-drawer__content {
+  z-index: var(--pf-c-drawer__content--ZIndex);
   flex-basis: var(--pf-c-drawer__content--FlexBasis);
   order: 0;
+  background-color: var(--pf-c-drawer__content--BackgroundColor);
+
+  &.pf-m-no-background {
+    --pf-c-drawer__content--BackgroundColor: transparent;
+  }
 }
 
 // Panel
 .pf-c-drawer__panel {
   position: relative;
+  z-index: var(--pf-c-drawer__panel--ZIndex);
   flex-basis: var(--pf-c-drawer__panel--FlexBasis);
   order: 1;
+  overflow: auto;
   background-color: var(--pf-c-drawer__panel--BackgroundColor);
+  box-shadow: var(--pf-c-drawer__panel--BoxShadow);
   transition-duration: var(--pf-c-drawer__panel--TransitionDuration);
   transition-property: var(--pf-c-drawer__panel--TransitionProperty);
   -webkit-overflow-scrolling: touch;
+
+  &::after {
+    position: absolute;
+    top: 0;
+    left: 0;
+    width: var(--pf-c-drawer__panel--after--Width);
+    height: 100%;
+    content: "";
+    background-color: var(--pf-c-drawer__panel--after--BackgroundColor);
+  }
+
+  &.pf-m-no-background {
+    --pf-c-drawer__content--BackgroundColor: transparent;
+  }
 }
 
-// Content body
-.pf-c-drawer__content-body.pf-m-padding {
-  padding: var(--pf-c-drawer__content-body--m-padding--PaddingTop) var(--pf-c-drawer__content-body--m-padding--PaddingRight) var(--pf-c-drawer__content-body--m-padding--PaddingBottom) var(--pf-c-drawer__content-body--m-padding--PaddingLeft);
+// Panel head
+.pf-c-drawer__head {
+  display: flex;
+  align-items: center;
 }
 
-// Panel body
-.pf-c-drawer__panel-body {
-  padding: var(--pf-c-drawer__panel-body--PaddingTop) var(--pf-c-drawer__panel-body--PaddingRight) var(--pf-c-drawer__panel-body--PaddingBottom) var(--pf-c-drawer__panel-body--PaddingLeft);
+// Modified content children
```

Comment:
Just an observation, but this is creating some pretty hefty selectors.

```
.pf-c-drawer__head,
.pf-c-drawer__header,
.pf-c-drawer__body {
  padding: var(--pf-c-drawer--child--PaddingTop) var(--pf-c-drawer--child--PaddingRight) var(--pf-c-drawer--child--PaddingBottom) var(--pf-c-drawer--child--PaddingLeft);
}

.pf-c-drawer__panel .pf-c-drawer__head:not(.pf-m-no-padding)+.pf-c-drawer__head,
.pf-c-drawer__panel .pf-c-drawer__header:not(.pf-m-no-padding)+.pf-c-drawer__head,
.pf-c-drawer__panel .pf-c-drawer__body:not(.pf-m-no-padding)+.pf-c-drawer__head,
.pf-c-drawer__panel .pf-c-drawer__head:not(.pf-m-no-padding)+.pf-c-drawer__header,
.pf-c-drawer__panel .pf-c-drawer__header:not(.pf-m-no-padding)+.pf-c-drawer__header,
.pf-c-drawer__panel .pf-c-drawer__body:not(.pf-m-no-padding)+.pf-c-drawer__header,
.pf-c-drawer__panel .pf-c-drawer__head:not(.pf-m-no-padding)+.pf-c-drawer__body,
.pf-c-drawer__panel .pf-c-drawer__header:not(.pf-m-no-padding)+.pf-c-drawer__body,
.pf-c-drawer__panel .pf-c-drawer__body:not(.pf-m-no-padding)+.pf-c-drawer__body {
  --pf-c-drawer--child--PaddingTop: 0;
}

.pf-c-drawer__content .pf-c-drawer__head.pf-m-padding+.pf-c-drawer__head.pf-m-padding,
.pf-c-drawer__content .pf-c-drawer__header.pf-m-padding+.pf-c-drawer__head.pf-m-padding,
.pf-c-drawer__content .pf-c-drawer__body.pf-m-padding+.pf-c-drawer__head.pf-m-padding,
.pf-c-drawer__content .pf-c-drawer__head.pf-m-padding+.pf-c-drawer__header.pf-m-padding,
.pf-c-drawer__content .pf-c-drawer__header.pf-m-padding+.pf-c-drawer__header.pf-m-padding,
.pf-c-drawer__content .pf-c-drawer__body.pf-m-padding+.pf-c-drawer__header.pf-m-padding,
.pf-c-drawer__content .pf-c-drawer__head.pf-m-padding+.pf-c-drawer__body.pf-m-padding,
.pf-c-drawer__content .pf-c-drawer__header.pf-m-padding+.pf-c-drawer__body.pf-m-padding,
.pf-c-drawer__content .pf-c-drawer__body.pf-m-padding+.pf-c-drawer__body.pf-m-padding {
  --pf-c-drawer--child--PaddingTop: 0;
}

.pf-c-drawer__content .pf-c-drawer__head,
.pf-c-drawer__content .pf-c-drawer__header,
.pf-c-drawer__content .pf-c-drawer__body {
  --pf-c-drawer--child--PaddingTop: 0;
  --pf-c-drawer--child--PaddingRight: 0;
  --pf-c-drawer--child--PaddingBottom: 0;
  --pf-c-drawer--child--PaddingLeft: 0;
}

.pf-c-drawer__head>.pf-c-drawer__head,
.pf-c-drawer__header>.pf-c-drawer__head,
.pf-c-drawer__body>.pf-c-drawer__head,
.pf-c-drawer__head>.pf-c-drawer__header,
.pf-c-drawer__header>.pf-c-drawer__header,
.pf-c-drawer__body>.pf-c-drawer__header,
.pf-c-drawer__head>.pf-c-drawer__body,
.pf-c-drawer__header>.pf-c-drawer__body,
.pf-c-drawer__body>.pf-c-drawer__body {
  --pf-c-drawer--child--PaddingTop: 0;
  --pf-c-drawer--child--PaddingRight: 0;
  --pf-c-drawer--child--PaddingBottom: 0;
  --pf-c-drawer--child--PaddingLeft: 0;
}

.pf-c-drawer__head.pf-m-no-padding,
.pf-c-drawer__header.pf-m-no-padding,
.pf-c-drawer__body.pf-m-no-padding {
  --pf-c-drawer--child--PaddingTop: 0;
  --pf-c-drawer--child--PaddingRight: 0;
  --pf-c-drawer--child--PaddingBottom: 0;
  --pf-c-drawer--child--PaddingLeft: 0;
}

.pf-c-drawer__head.pf-m-padding,
.pf-c-drawer__header.pf-m-padding,
.pf-c-drawer__body.pf-m-padding {
  --pf-c-drawer--child--PaddingTop: var(--pf-c-drawer--child--m-padding--PaddingTop);
  --pf-c-drawer--child--PaddingRight: var(--pf-c-drawer--child--m-padding--PaddingRight);
  --pf-c-drawer--child--PaddingBottom: var(--pf-c-drawer--child--m-padding--PaddingBottom);
  --pf-c-drawer--child--PaddingLeft: var(--pf-c-drawer--child--m-padding--PaddingLeft);
}
```

Reply:
Updated to this:

```
.pf-c-drawer__header,
.pf-c-drawer__body {
  padding: var(--pf-c-drawer--child--PaddingTop) var(--pf-c-drawer--child--PaddingRight) var(--pf-c-drawer--child--PaddingBottom) var(--pf-c-drawer--child--PaddingLeft); }
  .pf-c-drawer__header.pf-m-no-padding,
  .pf-c-drawer__body.pf-m-no-padding {
    --pf-c-drawer--child--PaddingTop: 0;
    --pf-c-drawer--child--PaddingRight: 0;
    --pf-c-drawer--child--PaddingBottom: 0;
    --pf-c-drawer--child--PaddingLeft: 0; }
  .pf-c-drawer__header.pf-m-padding,
  .pf-c-drawer__body.pf-m-padding {
    --pf-c-drawer--child--PaddingTop: var(--pf-c-drawer--child--m-padding--PaddingTop);
    --pf-c-drawer--child--PaddingRight: var(--pf-c-drawer--child--m-padding--PaddingRight);
    --pf-c-drawer--child--PaddingBottom: var(--pf-c-drawer--child--m-padding--PaddingBottom);
    --pf-c-drawer--child--PaddingLeft: var(--pf-c-drawer--child--m-padding--PaddingLeft); }
  .pf-c-drawer__header:not(.pf-m-no-padding) + *,
  .pf-c-drawer__body:not(.pf-m-no-padding) + * {
    --pf-c-drawer--child--PaddingTop: 0; }
```